tests/annotations/tests.py

```diff
@@ -148,6 +148,19 @@ def test_mixed_type_annotation_numbers(self):
         combined = int(test.pages + test.rating)
         self.assertEqual(b.combined, combined)
 
+    def test_empty_expression_annotation(self):
+        books = Book.objects.annotate(
+            selected=ExpressionWrapper(Q(pk__in=[]), output_field=BooleanField())
```

Comment:
Ok, let's leave the compiler changes and the test. There are enough people using this structure for us to either break it outright or support it. We can always fix any bugs that appear with `ExpressionWrapper(Q)`.

FWIW the problems I considered were to do with relabelling aliases or resolve_expression running multiple times. Now Q is just a resolvable, not really an expression. Whatever it returns must be a genuine expression. It returns `WhereNode` which has everything that an expression should have *except* `resolve_expression` and `output_field`.

If there's a way to trigger resolve_expression to be run multiple times, perhaps by using a queryset as a subquery, then it might be possible to trigger an error. I'm just brain dumping right now, I don't expect this to be tested for this patch.

Anyway, if we're going to all this trouble to support Q() objects as fields, then why not just add an `output_field = BooleanField()` to the WhereNode and bypass ExpressionWrapper altogether?